Src/Plugins/Security/SEC101.SecurePlaintextSecrets.json

```diff
@@ -165,6 +165,13 @@
           "MessageArguments": { "secretKind": "NPM API key" },
           "HelpUri": "https://docs.npmjs.com/creating-and-viewing-access-tokens"
         },
+        {
+          "Id": "SEC101/017",
+          "Name": "DoNotExposePlaintextSecrets/NpmAuthorToken",
+          "ContentsRegex": "$SEC101/017.NpmAuthorTokenV2",
```

Comment:
>NpmAuthorTokenV2 [](http://example.com/codeflow?start=40&length=16)

we should add a checksum validation for this secret kind, that's the substance of the token change (moving to v2). 

We need to separate this into a brand new rule, as we are likely to handle the 'identifiable' secrets differently (or someone needs to propose a convention for marking these as identifiable).

For now, try allocating a new rule id, SEC101/050 it appears? We should call the rule 'IdentifiableNpmAuthorToken' or 'NpmIdentifiableAuthorToken'. The former would sort all identifiable secret types together, the latter would keep the secret type bundled with the platform.

We could consider adding a new property denoting these as identifiable in adding to making it clear in the rule name. This would be valuable if we special-case behaviors in the engine itself (e.g., the engine could automatically elevate all 'identifiable' secrets to errors in the static analysis phase.

A future change, though, no need to worry about it now. Let's do handle the checksum. I just noticed we don't do this in the GH PAT either, so you'll need to follow up with me offline on what to do.

Reply:
Hello, we do the checksum in the internal version.

Reply:
I've split this out into a separate rule and added the supporting unit test and functional test files. Let me know what you think. I tried to organize the commits for easy reversal/changes.